File: codeforces/problemset/767A.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N;
    cin >> N;
    vector<int> A(N);
    for (auto &a : A)
        cin >> a;

    vector<bool> has(N + 1, false);
    int next = N;
    for (int n : A) {
        has[n] = true;
        while (has[next]) {
            cout << next << " ";
            next--;
        }
        cout << endl;
    }
    
    return 0;
}

File: codeforces/problemset/767A.py
N = int(input())
A = list(map(int, input().split()))

next = N
has = [False] * (N + 1)

for dayv in A:
    has[dayv] = True
    if dayv == next:
        while has[next]:
            print(next, end = ' ')
            next = next - 1
    print()
